### ISSUE TITLE
Embree version changes could cause an error in the dockerfile builds

### ISSUE THREAD

**Issue by Shimwell** [openmc-dev/openmc#1749]
Hi all

Just a preemptive issue notice as it look like embree is moving from version 3.12.1 to 3.12.2

This impact the Dockerfile which currently points to 3.12.1 in the double-down build

https://github.com/openmc-dev/openmc/blob/c262407af5eac38768910f4df7410ed08ae6ddda/Dockerfile#L111

I think that when the Dockerfile rebuild is triggered again that we will see a build error (I am seeing this in some other Dockerfile builds)

This can be temporarily fixed by changing 

```-DEMBREE_DIR=/embree/lib/cmake/embree-3.12.1 ; \```
to
```-DEMBREE_DIR=/embree/lib/cmake/embree-3.12.2 ; \```

This might be a pain to keep up to date so perhaps there are better ideas from those more familiar with the double-down build process @pshriwise

**Comment by pshriwise** [comment]
Thanks for staying ahead of the curve on this @Shimwell! I'm going to take the liberty of moving this to the [double-down](https://github.com/pshriwise/double-down/issues) repo though as it's really more about the interaction of Embree and that code.

**Comment by pshriwise** [comment]
Oh! I was a little hasty here. :facepalm: My apologies @Shimwell! Read this as double-down being incompatible with Embree 3.12.1...

In the simplest case, we could check out a specific version of Embree and stick with that. If there's good motivation to update the Embree version from time to time, then we can change it. I'll think about some ways to handle this more generally though.

**Comment by pshriwise** [comment]
It looks like Pixar studios parses the Embree version out of the `rtcore_version.h` file.

https://github.com/PixarAnimationStudios/USD/blob/release/cmake/modules/FindEmbree.cmake

I can look into doing something similar with double-down. Then we wouldn't have to worry about version numbers here.

**Comment by Shimwell** [comment]
> It looks like Pixar studios parses the Embree version out of the `rtcore_version.h` file.
> 
> https://github.com/PixarAnimationStudios/USD/blob/release/cmake/modules/FindEmbree.cmake
> 
> I can look into doing something similar with double-down. Then we wouldn't have to worry about version numbers here.

the ```find_library(EMBREE_LIBRARY``` you found looks like a perfect solution to me.

**Comment by pshriwise** [comment]
Alright @Shimwell, I've pushed [some updates](https://github.com/pshriwise/double-down/commit/cf01c4f019b52f3ce1015fc4556cf9e05e0b9c9c) that should allow us to remove the version numbers in the Dockerfile here.

**Comment by Shimwell** [comment]
Great does that mean we can change these lines in the docker file https://github.com/openmc-dev/openmc/blob/c262407af5eac38768910f4df7410ed08ae6ddda/Dockerfile#L110-L111

from 
```
             -DMOAB_DIR=/usr/local \
             -DEMBREE_DIR=/embree/lib/cmake/embree-3.12.1 ; \
```
to
```
             -DMOAB_DIR=/usr/local ; \
```

If so that is a really nice upgrade which simplifies things for us, thanks for all the work upstream

**Comment by pshriwise** [comment]
We'll still need the `-DEMBREE_DIR` argument for now, but it can be reduced to the root directory of the Embree install (`-DEMBREE_DIR=/embree` in this case) with no dependence on the version number. Although if `/` is in CMake's default search path for packages maybe we don't need to specify it at all? Let me know what you find out! :)

**Comment by Shimwell** [comment]
Confirming that ...

 ```-DEMBREE_DIR=/embree && \``` works nicely
https://github.com/ukaea/openmc_workshop/actions/runs/506203033

removing ```-DEMBREE_DIR``` entirely from the args doesn't work 
https://github.com/ukaea/openmc_workshop/actions/runs/506283745

This is a tiny change but I am happy to throw in a PR if that helps

**Comment by pshriwise** [comment]
PR's with tiny changes are my favorite to review :smile: If you're willing, that would be great.

**Comment by pshriwise** [comment]
This was resolved by #1750
